lib/src/core/message/guild_emoji.dart

```diff
@@ -12,29 +14,43 @@ abstract class IBaseGuildEmoji implements SnowflakeEntity, IEmoji {
   bool get isPartial;
 
   /// Returns cdn url to emoji
-  String get cdnUrl;
+  String cdnUrl({String? format = "webp", int? size = 128});
+
+  /// The name of the emoji.
+  String get name;
+
+  /// Whether this emoji is animated.
+  bool get animated;
 
   /// Creates partial emoji from given String or Snowflake.
-  factory IBaseGuildEmoji.fromId(Snowflake id) => GuildEmojiPartial(id);
+  factory IBaseGuildEmoji.fromId(Snowflake id) => GuildEmojiPartial({"id": id});
 }
 
 abstract class BaseGuildEmoji extends SnowflakeEntity implements IBaseGuildEmoji {
   /// True if emoji is partial.
   @override
   bool get isPartial;
 
+  /// Whether this emoji is animated.
+  @override
+  bool get animated;
+
   /// Returns cdn url to emoji
   @override
-  String get cdnUrl => "https://cdn.discordapp.com/emojis/$id.png";
+  String cdnUrl({String? format = "webp", int? size = 128}) => "${Constants.cdnUrl}/emojis/$id.$format?size=$size";
```

Comment:
format shouldn;t be null because if someone passes here null it would evaulable to 'null' in resulting string which is garbage.

Instead would be better in that case I think to leave both as nullable without value and if null provided for format then you automatically should check if emoji is animated or not and if user provided non null format then it should be used that. and whole `size` get param should be any appended to string if `size` param is not null

Something like:
```dart
String cdnUrl({String? format = "webp", int? size = 128}) {
  var url = "${Constants.cdnUrl}/emojis/$id.";

  if (format == null) {
    if (animated) {
      url += "gif";
    } else {
      url += "webp";
    }
  } else {
    url += format;
  }

  if (size != null) {
    url += "?size=$size";  
  }

  return url;
}
```

Reply:
I agree with that, but, shouldn't we let the developer choose if an animated emoji will be displayed as a static/dynamic image?

Reply:
yeah, my example includes that case

Reply:
Like:
```dart
String cdnUrl({String? format = "webp", int? size = 128, bool? animate = false}) {
	var url = "${Constants.cdnUrl}/emojis/$id.";
	
	if (format == null) {
		if(animated && animate != null && animate) {
			url += "gif";
		} else {
			url += "webp";
		}
	} else {
		if (animated && animate != null && animate) {
			url += "gif";
		} else {
			url += format;
		}
	}

	if (size != null) {
		url += "?size=$size";
	}
	
	return url;
}
```

Reply:
why do we need `animated` checks? If developer wants static emoji even if emoji is animated we should give him non animated url. So my version stands up:
```dart
String cdnUrl({String? format, int? size}) {
  var url = "${Constants.cdnUrl}/emojis/$id.";

  if (format == null) {
    if (animated) {
      url += "gif";
    } else {
      url += "webp";
    }
  } else {
    url += format;
  }

  if (size != null) {
    url += "?size=$size";  
  }

  return url;
}
```

Reply:
Oh, that's right, didn't pay attention